### PR TITLE
CLN: misc cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**
Things I find myself doing in multiple branches.

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -2279,7 +2279,7 @@ def _from_factorized(cls, uniques, original):
             original.categories.take(uniques), dtype=original.dtype
         )
 
-    def equals(self, other):
+    def equals(self, other) -> bool:
```
Comment (WillAyd) [inline pandas/core/arrays/categorical.py]:
```suggestion
    def equals(self, other: Any) -> bool:
```

Purely stylistic but we've been typing `other` as Any elsewhere, so good to do here as well (applicable in a few spots)

**Review by WillAyd — CHANGES_REQUESTED** [review]
minor comment otherwise lgtm

File: pandas/core/arrays/categorical.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py]: the docstring says other is Categorical; id prefer not to introduce mismatches

**Comment by jbrockmendel** [comment]
reverted annotations to push the other parts down the field

**Comment by jbrockmendel** [comment]
@WillAyd gentle ping; reverted the parts that were non-trivial

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

File: pandas/core/indexes/datetimelike.py [hunk 2]
```diff
@@ -156,13 +156,11 @@ def equals(self, other) -> bool:
...
         try:
             res = self.get_loc(key)
-            return (
-                is_scalar(res)
-                or isinstance(res, slice)
-                or (is_list_like(res) and len(res))
-            )
         except (KeyError, TypeError, ValueError):
             return False
+        return bool(
```
Comment (jreback) [inline pandas/core/indexes/datetimelike.py]: isn't the bool redundant here?

Comment (jbrockmendel) [reply pandas/core/indexes/datetimelike.py]: not according to mypy

**Pull request merged**